garak/resources/autodan/genetic.py

```diff
@@ -2,33 +2,54 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import gc
+import nltk.downloader
 import numpy as np
 import torch
 import random
 import openai
-import os
 import re
 import nltk
 from nltk.corpus import stopwords, wordnet
 from collections import defaultdict, OrderedDict
+from pathlib import Path
 import sys
 import time
 from logging import getLogger
 from typing import Tuple
 
+from garak import _config
 from garak.resources.autodan.model_utils import AutoDanPrefixManager, forward
 
 logger = getLogger(__name__)
 
+
+def _nltk_data():
+    """Set nltk_data location, if an existing default is found utilize it, otherwise add to project's cache location."""
+    from nltk.downloader import Downloader
+
+    default_path = Path(Downloader().default_download_dir())
+    if not default_path.exists():
+        # if path not found then place in the user cache
+        # get env var for NLTK_DATA, fallback to create in cachedir / nltk_data
+        logger.debug("nltk_data location not found using project cache location")
+        _nltk_data_path.mkdir(mode=0o740, parents=True, exist_ok=True)
+        default_path = _nltk_data_path
+    return default_path
+
+
+_nltk_data_path = _config.transient.cache_dir / "nltk_data"
```

Comment:
Should this path designate this as a `resource` as output files from `gpg` and `beast` do?

```suggestion
_nltk_data_path = _config.transient.cache_dir / "resources" / "nltk_data"
```

Reply:
would prefer to let `nltk` manage its own data here, to avoid duplicating `nltk` data.

Reply:
`nltk` is still managing the location here, I am simply adding an additional location if no `nltk` is already on the host.  This ensure that a host that is only for `garak` consolidates the data it downloads if the configuration of the host does not already provide an existing cache.